app/main.py

```diff
@@ -3,25 +3,43 @@
 import os
 from collections.abc import AsyncIterator
 from contextlib import asynccontextmanager
-from typing import Annotated, Never
+from typing import Annotated, Any, AsyncContextManager, Callable, Mapping, Never
 
 from fastapi import Depends, FastAPI, status
 from fastapi.staticfiles import StaticFiles
 from loguru import logger
 from sqlalchemy import select
 from sqlalchemy.ext.asyncio import AsyncSession, async_scoped_session
 
-from app.db.db_init import close_db, get_read_session_with_depends, get_write_session_with_depends, init_db
+from app.db.db_init import (
+    close_db,
+    get_read_session_with_depends,
+    get_write_session_with_depends,
+    init_db,
+)
 from app.legacy.v2.notifications.rest import v2_notification_router
 from app.logging.logging_config import CustomizeLogger
-from app.providers.provider_aws import ProviderAWS
+from app.state import ENPState
 from app.v3 import api_router as v3_router
 
 MKDOCS_DIRECTORY = 'site'
 
 
+class CustomFastAPI(FastAPI):
+    """Custom FastAPI class to include ENPState."""
+
+    def __init__(self, lifespan: Callable[['CustomFastAPI'], AsyncContextManager[Mapping[str, Any]]]) -> None:
```

Comment:
I understand we're aiming to avoid using Any. However, given that the context includes resources like the database, application state (str | dicts| bools ), and potentially other additions in the future, it’s  difficult to define a fully specific Types upfront. 

Or maybe I'm misunderstanding `asynccontextmanager` and how we're implementing `def lifespan`.

Reply:
Long term, it might make sense to look into TypedDict for mappings like that https://peps.python.org/pep-0589/

Database data and different subsets of app state could then be composed by combining the smaller TypedDict defs.

Reply:
Ya. Maybe in the future we could build out custom dictionary objects to handle typing for lifespan. Feel like too much at the moment.